### ISSUE TITLE
[BUG?] Client certificate issue

### ISSUE THREAD

**Issue by hcgraf** [astubenbord/paperless-mobile#216]
First, thanks for the great app, which made me aware of the existence of `paperless-ngx` (I was still on `paperless-ng`) :)
This might be a bug or a misconfiguration from my side – in that case some guidelines and maybe a documentation-improvement would be great!

**Describe the bug**
When using a client certificate, I see the following two (likely related) issues:
1. "Test connection" doesn't seem to terminate (green progress bar keeps spinning, no result shown)
2. "Continue" & entering username/password is still possible, but login fails with `type 'Null' is not a subtype of type 'String'`

**To Reproduce**
1. Go to 'Add Account'
3. Enter your server address
4. Select certificate file & enter password
5. No "connection test" result is shown, progress bar keeps spinning
6. Click continue
7. Enter password & username, click login
8. Error message shown: `type 'Null' is not a subtype of type 'String'`

**Expected behavior**
Connection test & login works

**Device Information:**
 - Device: Xiaomi Mi A3
 - OS: Android 11
 - Paperless Mobile version: 2.3.8+1043
 - Paperless Server version: v1.16.2 (API v3)

**Additional context**
Paperless-ngx server running in Docker behind an Apache reverse-proxy, using pretty exactly [this](https://github.com/paperless-ngx/paperless-ngx/wiki/Using-a-Reverse-Proxy-with-Paperless-ngx#apache) configuration.

Additional client certificate requirement:
```
SSLVerifyClient require
SSLVerifyDepth 1
SSLCACertificateFile /etc/apache2/ca/ca.cert.pem
```
Client certificate created like this, and confirmed working in a browser:
```
openssl genrsa -out private/paperless.key.pem 2048
openssl req -config openssl.conf -key private/paperless.key.pem -new -sha256 -out csr/paperless-client.csr.pem
openssl ca -config openssl.conf -extensions usr_cert -days 365 -notext -md sha256 -in csr/paperless-client.csr.pem -out certs/paperless-client.cert.pem
openssl pkcs12 -export -inkey private/paperless.key.pem -in certs/paperless-client.cert.pem -out paperless-client.pfx
```
Questions / observations
* Does the **key** (`pem` file) need to be encrypted (`-aes256`) or not?
* Does the **export** (`pfx` file) need to be encrypted (non-empty password during export)?
* Is the `passphrase` in the App for the **key** or `pfx`-encryption?
* From the error, it looks like the App either doesn't find a required information in the `pfx` file, or isn't able to decrypt it.
* Could it be an Android permission issue? I granted everything, but is says "only media access". So I'm wondering if the App is only allowed to list/select the `pfx` file – but not actually read it… Any way to confirm something like that?
* Why are you using the `pfx` file directly, and not a client certificate from the Android certificate storage (I know of the open-source clients for `openHAB` and `HomeAssistant` which are doing it that way, if you need inspiration ;) )

Thanks a lot for any pointers, I'm happy to help with further information / debugging :)

**Comment by astubenbord** [comment]
Any chance you could get the logs from the app? 

This can be achieved either through `flutter logs` or `adb logcat | grep flutter`. I assume it's an error in some model where a nullable field is assumed non nullable.

**Comment by hcgraf** [comment]
Wow, that was quick :) 
I used `logcat` on the device itself, with `flutter` as the filter.
The error is on line `42`, `135` and `156` – hope that helps :) 
[logcat_06-19-2023_19-37-37.txt](https://github.com/astubenbord/paperless-mobile/files/11790782/logcat_06-19-2023_19-37-37.txt)

**Comment by hcgraf** [comment]
Hi again,

I got it to work. The most relevant part seems to be:
When entering the server address **do not use one from the drop-down proposals** or type at least one letter manually.
Maybe also helpful: select the certificate and enter the passphrase **before** entering the server address.

This is certainly undesired behavior, but at least a workaround for me. And is should be easier to find the problems for somebody familiar with the code :)

**Comment by astubenbord** [comment]
I see, maybe the input is not stored in the form. I'll check. Also, the suggested host feature is quite new and I didn't test it much. I'll do some more testing and see if I can fix it. Thanks

**Comment by hcgraf** [comment]
Thanks a lot! Let me know if I can help testing anything!

**Comment by WGandy** [comment]
I was running into what appears to be the same problem with my instance that is behind Caddy with let's encrypt certs. The _other_ paperless android app would log in but not this one. But, as alluded to above, if I retype the address instead of selecting what appears to be the exact same address from the saved "pull down" it allowed me to log in.  

Now if I can just figure out why neither of the android apps would login into my paperless-ngx instance when it's been routed through haproxy via sni filtering and then to caddy where the TLS is terminated, then I would be in business! (I can log in from a browser with this config, but not the mobile apps).

**Comment by astubenbord** [comment]
Does caddy perform a redirect?

**Comment by astubenbord** [comment]
Should be fixed in v3.0.1.
